registration/manager_test.go

```diff
@@ -0,0 +1,221 @@
+package registration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/hashicorp/consul-terraform-sync/config"
+	"github.com/hashicorp/consul-terraform-sync/logging"
+	mocks "github.com/hashicorp/consul-terraform-sync/mocks/client"
+	consulapi "github.com/hashicorp/consul/api"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewRegistrationManager(t *testing.T) {
+	testcases := []struct {
+		name            string
+		conf            *RegistrationManagerConfig
+		expectedService *service
+	}{
+		{
+			"defaults",
+			&RegistrationManagerConfig{
+				ID:               "cts-123",
+				Port:             123,
+				TLSEnabled:       false,
+				SelfRegistration: config.DefaultSelfRegistrationConfig(),
+			},
+			&service{
+				name:      defaultServiceName,
+				tags:      defaultServiceTags,
+				id:        "cts-123",
+				port:      123,
+				namespace: "",
+			},
+		},
+		{
+			"namespace",
+			&RegistrationManagerConfig{
+				ID:         "cts-123",
+				Port:       123,
+				TLSEnabled: false,
+				SelfRegistration: &config.SelfRegistrationConfig{
+					Namespace: config.String("ns-1"),
+				},
+			},
+			&service{
+				name:      defaultServiceName,
+				tags:      defaultServiceTags,
+				id:        "cts-123",
+				port:      123,
+				namespace: "ns-1",
+			},
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			client := new(mocks.ConsulClientInterface)
+			m := NewRegistrationManager(tc.conf, client)
+
+			// Verify general attributes
+			assert.NotNil(t, m)
+			assert.Equal(t, m.client, client)
+			assert.NotNil(t, m.logger)
+
+			// Verify service attributes
+			assert.NotNil(t, m.service)
+			assert.Equal(t, tc.expectedService.name, m.service.name)
+			assert.Equal(t, tc.expectedService.id, m.service.id)
+			assert.Equal(t, tc.expectedService.tags, m.service.tags)
+			assert.Equal(t, tc.expectedService.port, m.service.port)
+			assert.Equal(t, tc.expectedService.namespace, m.service.namespace)
+
+			// Verify service has default health check
+			assert.NotNil(t, m.service.checks)
+			assert.Equal(t, 1, len(m.service.checks))
+			check := m.service.checks[0]
+			defaultCheck := defaultHTTPCheck(tc.conf)
+			assert.Equal(t, defaultCheck, check)
```

Comment:
Wondering if it's possible to consolidate some of these asserts to something like:

```
assert.NotNil(t, m.service)
tc.expectedService.checks = []*consulapi.AgentServiceCheck{defaultHTTPCheck(tc.conf)}
assert.Equal(t, tc.expectedService, m.service)
```

Maybe it's not very extensible for future test cases though?

Reply:
I like your consolidation, I will update!